src/System.Reflection.TypeExtensions/tests/EventInfoTests.cs

```diff
@@ -0,0 +1,184 @@
+            Assert.Throws(exceptionType, () => eventInfo.RemoveEventHandler(target, handler));
+        }
+
+        [Theory]
+        [InlineData("PublicEvent")]
+        [InlineData("Public_Event")]
+        [InlineData("Private_Event")]
+        [InlineData("PrivateEvent")]
+        [InlineData("ProtectedEvent")]
+        [InlineData("ProtectedInternalEvent")]
+        [InlineData("ProtectedInternalEvent")]
+        public void Attributes_IsSpecialName(string name)
+        {
+            EventInfo eventInfo = Helpers.GetEvent(typeof(EI_Class), name);
+            Assert.Equal(EventAttributes.None, eventInfo.Attributes);
+            Assert.False(eventInfo.IsSpecialName);
+        }
+
+        [Theory]
+        [InlineData("PublicEvent", typeof(VoidDelegate))]
+        [InlineData("PrivateEvent", typeof(VoidDelegate))]
+        [InlineData("ProtectedEvent", typeof(VoidDelegate))]
+        [InlineData("ProtectedInternalEvent", typeof(VoidDelegate))]
+        public void EventHandlerType(string name, Type expected)
+        {
+            EventInfo eventInfo = Helpers.GetEvent(typeof(EI_Class), name);
+            Assert.Equal(expected, eventInfo.EventHandlerType);
+        }
+
+        [Theory]
+        [InlineData("PublicEvent", "Void add_PublicEvent(System.Reflection.Tests.VoidDelegate)", "Void add_PublicEvent(System.Reflection.Tests.VoidDelegate)")]
+        [InlineData("PrivateEvent", null, "Void add_PrivateEvent(System.Reflection.Tests.VoidDelegate)")]
+        [InlineData("ProtectedEvent", null, "Void add_ProtectedEvent(System.Reflection.Tests.VoidDelegate)")]
+        [InlineData("ProtectedInternalEvent", null, "Void add_ProtectedInternalEvent(System.Reflection.Tests.VoidDelegate)")]
+        public void GetAddMethod(string name, string publicToString, string nonPublicToString)
+        {
+            EventInfo eventInfo = Helpers.GetEvent(typeof(EI_Class), name);
+            MethodInfo method = eventInfo.GetAddMethod();
+            Assert.Equal(publicToString == null, method == null);
+            Assert.Equal(publicToString, method?.ToString());
+
+            method = eventInfo.GetAddMethod(false);
+            Assert.Equal(publicToString == null, method == null);
+            Assert.Equal(publicToString, method?.ToString());
+
+            method = eventInfo.GetAddMethod(true);
+            Assert.NotNull(method);
+            Assert.Equal(nonPublicToString, method.ToString());
+        }
+
+        [Theory]
+        [InlineData("PublicEvent")]
+        [InlineData("PrivateEvent")]
+        [InlineData("ProtectedEvent")]
+        [InlineData("ProtectedInternalEvent")]
+        public void GetRaiseMethod(string name)
+        {
+            EventInfo eventInfo = Helpers.GetEvent(typeof(EI_Class), name);
+            Assert.Null(eventInfo.GetRaiseMethod());
+            Assert.Null(eventInfo.GetRaiseMethod(false));
+            Assert.Null(eventInfo.GetRaiseMethod(true));
```

Comment:
Not for this PR, which I understand is fixing up the existing tests, but it looks like we don't have any tests which validate a non-null raise method.  We'd likely need to write such a raise handler in IL.  Pretty much no one uses raise handlers, though, as neither C# nor Visual Basic produces them (I think C++/CLI does), so this is low priority.